signer/src/json.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright 2019-2024 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+//! A Polkadot-JS account loader.
+
+use base64::Engine;
+use core::fmt::Display;
+use crypto_secretbox::{
+    aead::{Aead, KeyInit},
+    Key, Nonce, XSalsa20Poly1305,
+};
+use serde::Deserialize;
+use subxt_core::utils::AccountId32;
+
+use crate::sr25519;
+
+/// Error
+#[derive(Debug)]
+pub enum Error {
+    /// The keypair has an unsupported encoding.
+    UnsupportedEncoding,
+    /// Base64 decoding error.
+    Base64(base64::DecodeError),
+    /// Wrong Scrypt parameters
+    UnsupportedScryptParameters {
+        /// N
+        n: u32,
+        /// p
+        p: u32,
+        /// r
+        r: u32,
+    },
+    /// Decryption error.
+    Secretbox(crypto_secretbox::Error),
+    /// The decrypted keys are not valid.
+    InvalidKeys,
+}
+
+impl_from!(base64::DecodeError => Error::Base64);
+impl_from!(crypto_secretbox::Error => Error::Secretbox);
+
+impl Display for Error {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            Error::UnsupportedEncoding => write!(f, "Unsupported encoding."),
+            Error::Base64(e) => write!(f, "Base64 decoding error: {e}"),
+            Error::UnsupportedScryptParameters { n, p, r } => {
+                write!(f, "Unsupported Scrypt parameters: N: {n}, p: {p}, r: {r}")
+            }
+            Error::Secretbox(e) => write!(f, "Decryption error: {e}"),
+            Error::InvalidKeys => write!(f, "The decrypted keys are not valid."),
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for Error {}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct EncryptedJsonDescriptor {
```

Comment:
this name is also quite confusing I would just call it EncryptionMetadata

Reply:
done